screens/signUp.js

```diff
@@ -28,21 +29,30 @@ const SignUp = () => {
             style={tailwind("mb-4")}
             placeholder="name"
             icon={require("../assets/icons/black/person.png")}
-            onChange={(e) => setName(e)}
+            ref={name}
+            onChange={(e) => {
```

Comment:
This looks better! But, it looks like you are still updating the reference with every change, which is similar to the previous issue. Instead, we should not use the `onChange` method and instead access the reference when we submit; when the signup button has been pressed.

So when the login button is pressed and you call the API endpoint, then that's when you want to get the current value of all of the inputs, not every time your input a character.